core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -216,19 +223,26 @@ object ConsoleConsumer extends Logging {
     * are conflicting.
     */
   def setAutoOffsetResetValue(config: ConsumerConfig, props: Properties) {
+    val (earliestConfigName, latestConfigName) = if (config.useOldConsumer)
+      (OffsetRequest.SmallestTimeString, OffsetRequest.LargestTimeString)
+    else
+      ("earliest", "latest")
+
     if (props.containsKey(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG)) {
       // auto.offset.reset parameter was specified on the command line
-      if (config.options.has(config.resetBeginningOpt) && "latest".equals(props.getProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG))) {
+      if (config.options.has(config.resetBeginningOpt) && earliestConfigName != props.getProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG)) {
```

Comment:
Can `props.getProperty` return null? We probably don't want to error in that case, right? Changing the message as per your suggestion sounds good.